### PR TITLE
📖 [Docs] Add support for skip ranges and replacements

### LINKED ISSUE #744: Docs: Implement full support for replaces, skips, and skipRange (and make default)
# User story

As an admin, when I have an installed ClusterExtension, I expect that the upgrade edges defined in the catalog (i.e. replaces, skips, and skipRange) will be honored.

## List the user-facing issues in the epic that require documentation.

- Issue: #738 (single issue epic)

## Does this epic require new content or updates to existing content?

- [ ] Net-new content
- [x] Update existing content

My assumption is that there are already docs that talk about how OLMv1 uses semver semantics when assessing upgrade successors in a channel. If that's true, those docs need to be updated to talk about replaces, skips, and skipRange.

Even with this new support for replaces, skips, and skipRange, OLMv1 logic is subtly different:
- In OLMv0 if there are multiple possible successors, the one closest to the channel head is chosen
- In OLMv1 if there are multiple possible successors, the one with the highest semver version is chosen.

## What type of documentation is required for this epic?

- [ ] Release note
- [ ] Concept
- [ ] Procedure
- [ ] Reference
- [x] Not sure

## Is there prior art or reference materials that would help write the documentation?

https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/

Documentation should be contributed to the [`docs/drafts/` directory](https://github.com/operator-framework/operator-controller/tree/main/docs/drafts)

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Closes #744

@perdasilva: I am not sure if you are using this file as part of your work. I just wanted to make sure we aren't stepping on each other's toes :)

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/drafts/upgrade-support.md [hunk 1]
```diff
@@ -1,33 +1,58 @@
...
+kind: ClusterExtension
+metadata:
+  name: <extension_name>
+spec:
+  installNamespace: <namespace>
+  packageName: <package_name>
+  serviceAccount:
+    name: <service_account>
+  upgradeConstraintPolicy: Ignore
+  version: "<version_or_version_range>"
```
Comment (joelanford) [inline docs/drafts/upgrade-support.md:46]: Non-blocking (I think we'll need to sweep the entire repo once all the API changes are finished): All these fields are getting moved around: 
- This example is not currently valid.
- Even if you made it valid according to what is in `main` right now, there are other open PRs that will invalidate it.

File: docs/drafts/upgrade-support.md [hunk 2]
```diff
@@ -1,33 +1,58 @@
...
 ```
 
-In a future release, it is planned to remove the `ForceSemverUpgradeConstraints` feature gate and allow package authors to specify upgrade constraint semantics at the catalog level.
+where setting the `upgradeConstraintPolicy` to:
+
+`Ignore`
+:   Does not limit the next version to the set of successors, and instead allows for any downgrade, sidegrade, or upgrade.
+
+`Enforce`
```
Comment (joelanford) [inline docs/drafts/upgrade-support.md:54]: Another upcoming change: https://github.com/operator-framework/operator-controller/issues/1151

File: docs/drafts/upgrade-support.md (hunk 1 above)
Comment (michaelryanpeter) [reply docs/drafts/upgrade-support.md:46]: With Material MK Docs, we can actually embed YAML files into a markdown page.

Maybe we can create an issue right before or after 1.0.0 GA to switch our inline examples to embedded YAML files? That way we can update our examples all in one go.

Comment (michaelryanpeter) [reply docs/drafts/upgrade-support.md:46]: It looks like #1200 has the relevant doc changes queued up in that PR.

File: docs/drafts/upgrade-support.md (hunk 2 above)
Comment (michaelryanpeter) [reply docs/drafts/upgrade-support.md:54]: @joelanford Is this something we would want to change to the new value now? Or do we want to update #1151 with a comment or a docs issue?

Comment (joelanford) [reply docs/drafts/upgrade-support.md:54]: Use what you have, which is what is on `main`. The PR for that change should hopefully find/replace these.

File: docs/drafts/upgrade-support.md (hunk 1 above)
Comment (joelanford) [reply docs/drafts/upgrade-support.md:46]: #1200 merged, which makes this section incorrect if it merges. I'm happy to merge this and fix in a follow up (hence my approval on the PR), but I'll defer to you if you want to go ahead and fix now.

**Review by joelanford — APPROVED** [review]

**Pull request merged**